src/webui/www/private/scripts/torrent-content.js

```diff
@@ -0,0 +1,572 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+"use strict";
+
+window.qBittorrent ??= {};
+window.qBittorrent.TorrentContent ??= (() => {
+    const exports = () => {
+        return {
+            init: init,
+            normalizePriority: normalizePriority,
+            isFolder: isFolder,
+            createDownloadCheckbox: createDownloadCheckbox,
+            updateDownloadCheckbox: updateDownloadCheckbox,
+            createPriorityCombo: createPriorityCombo,
+            updatePriorityCombo: updatePriorityCombo,
+            updateData: updateData,
+            clearFilterInputTimer: clearFilterInputTimer
+        };
+    };
+
+    let torrentFilesTable;
+    const FilePriority = window.qBittorrent.FileTree.FilePriority;
+    const TriState = window.qBittorrent.FileTree.TriState;
+    let torrentFilesFilterInputTimer = -1;
+    let onFilePriorityChanged = null;
+
+    const normalizePriority = (priority) => {
+        priority = Number(priority);
+
+        switch (priority) {
+            case FilePriority.Ignored:
+            case FilePriority.Normal:
+            case FilePriority.High:
+            case FilePriority.Maximum:
+            case FilePriority.Mixed:
+                return priority;
+            default:
+                return FilePriority.Normal;
+        }
+    };
+
+    const triStateFromPriority = (priority) => {
+        switch (normalizePriority(priority)) {
+            case FilePriority.Ignored:
+                return TriState.Unchecked;
+            case FilePriority.Normal:
+            case FilePriority.High:
+            case FilePriority.Maximum:
+                return TriState.Checked;
+            case FilePriority.Mixed:
+                return TriState.Partial;
+        }
+    };
+
+    const isFolder = (fileId) => {
+        return fileId === -1;
+    };
+
+    const getAllChildren = (id, fileId) => {
+        const getChildFiles = (node) => {
+            rowIds.push(node.data.rowId);
+            fileIds.push(node.data.fileId);
```

Comment:
`rowIds` and `fileIds` are unbounded when `getChildFiles` is declared. I would declare `rowIds` and `fileIds` before this function to avoid this situation. The old code wasn't like this.

Reply:
This is the same as the original code - just moved.

Reply:
>This is the same as the original code - just moved.

Doesn't look like the original code. It reorganized the code.
https://github.com/qbittorrent/qBittorrent/blob/ae8a6689dc1451ae5d8e7a717d5c0a0f1056cb9d/src/webui/www/private/scripts/prop-files.js#L64-L96